cohortextractor/main.py

```diff
@@ -74,19 +163,42 @@ def load_cohort(definition_path):
     return cohort_classes[0]
 
 
-def load_module(definition_path):
-    definition_dir = definition_path.parent
-    module_name = definition_path.stem
+def load_module(definition_path, index_date=None):
+    if index_date is not None:
+        with temp_definition_path(definition_path, index_date) as temp_path:
+            with open(definition_path, "r") as orig, open(temp_path, "w") as new:
+                # Copy the cohort definition file, replace the BASE_INDEX_DATE definition
+                # with the current index date and load this modified module
+                contents = orig.read()
+                pattern = r"(BASE_INDEX_DATE\s*=\s*.+)([\s|\n].*)"
+                contents = re.sub(
+                    pattern, rf'BASE_INDEX_DATE = "{index_date}"\2', contents
+                )
+                new.write(contents)
+            module = _load_module(temp_path)
+        return module
+    return _load_module(definition_path)
```

Comment:
I found this a bit hard to follow with the nesting. What if the `None` case was handled as a guard clause? (If it is `None` then return immediately.)